web/src/components/storage/utils.ts

```diff
@@ -257,6 +277,34 @@ const reuseDevice = (volume: Volume): boolean =>
 const volumeLabel = (volume: Volume): string =>
   volume.mountPath === "/" ? "root" : volume.mountPath;
 
+/**
+ * @see filesystemType
+ */
+const filesystemName = (fstype: string): string => {
+  const name = FILESYSTEM_NAMES[fstype];
+
+  // eslint-disable-next-line agama-i18n/string-literals
+  if (name) return _(name);
+
+  // Fallback for unknown filesystem types
+  return (fstype.charAt(0).toUpperCase() + fstype.slice(1));
+}
+
+/**
+ * String to represent the filesystem type
+ *
+ * @returns undefined if there is not enough information
+ */
+const filesystemType = (filesystem: configModel.Filesystem): string | undefined=> {
+  if (filesystem.type) {
+    if (filesystem.snapshots) return _("Btrfs with snapshots");
+
+    return filesystemName(filesystem.type);
+  }
+
+  return undefined;
```

Comment:
I'm fine with that explicit return. But if you drop it and function reach this point it will return undefined too.